### PR TITLE
feat: v6 deployment improvements

### PR DESCRIPTION
**Pull request by milapsheth**
AXE-1752

See README

### CODE REVIEW

File: evm/README.md [hunk 1]
```diff
@@ -18,7 +18,16 @@ You also need to create a `info/keys.json` file. See `evm/.example.keys.json` fo
 4. Add additional params in `.env` such as admin addresses, governance etc.
 5. If you'd like to auto-verify the contract on the explorer, then add the explorer API key under `info/keys.json` (see `.example.keys.json`), and add `--verify` flag
 6. Run the following depending on the service,
-`node evm/deploy-gateway-v5.x.js --env testnet -n fantom`
+`node evm/deploy-gateway-v5.0.x.js --env testnet -n fantom`
+
+#### Gateway Upgrade
```
Comment (milapsheth) [inline evm/README.md:23]: This covers v6.1 -> v6.1 upgrade. v4.3 -> v6.1 upgrade is similar (need to deploy v4.3 gateway first).

A subsequent update will cover the upgrade tx when using InterchainGovernance

File: evm/deploy-gateway-v5.0.x.js [hunk 2]
```diff
@@ -68,126 +99,152 @@ async function deploy(config, options) {
...
+        await auth.deployTransaction.wait(chain.confirmations);
 
         contractsToVerify.push({
             address: auth.address,
             params: [params],
         });
     }
 
-    if (reuseProxy) {
+    if (reuseProxy && reuseHelpers) {
```
[the remark this answers is not in the document]
Comment (milapsheth) [reply evm/deploy-gateway-v5.0.x.js:150]: i'd like to keep auth and token deployer separate. it's more convenient if trying out some modifications locally

File: evm/deploy-gateway-v5.0.x.js [hunk 3]
```diff
@@ -196,50 +253,114 @@ async function deploy(config, options) {
...
+        printInfo('Verified all contracts!');
+    }
+}
+
+async function upgrade(config, options) {
+    const { chainName, privateKey, yes } = options;
+
+    const contractName = 'AxelarGateway';
+
+    const chain = config.chains[chainName] || { contracts: {}, name: chainName, id: chainName, tokenSymbol: 'ETH' };
```
[the remark this answers is not in the document]
Comment (milapsheth) [reply evm/deploy-gateway-v5.0.x.js:301]: it's expected to pass the rpc via options for the default object case, so throwing an error is fine if it wasn't defined. added to default object regardless for simplicity

**Review by deanamiel — APPROVED** [review]

**Comment by milapsheth** [comment]
merging for now to unblock other PRs

**Pull request merged**